site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+    if (call.selected_field !== 'issue_type') {
+        const callResponse: AppCallResponse = {
+            type: 'error',
+            error: `Unexpected refresh field: ${call.selected_field}`,
+        };
+
+        res.json(callResponse);
+        return;
+    }
+
+    const baseFields: AppField[] = [
+        {
+            name: 'summary',
+            modal_label: 'Summary',
+            type: 'text',
+            value: values.summary, // We're using `values.summary` here to make sure we keep the user's entered values in tact when we return this new form
+            is_required: true,
+            min_length: 10,
+        },
+        {
+            name: 'description',
+            modal_label: 'Description',
+            type: 'text',
+            subtype: 'textarea',
+            value: values.description, // Same for this field
+        },
+        {
+            name: 'issue_type',
+            modal_label: 'Issue Type',
+            type: 'static_select',
+            value: values.issue_type, // And this field
+            refresh: true,
+            options: [
+                {label: 'Story', value: 'story'},
+                {label: 'Task', value: 'task'},
+                {label: 'Bug', value: 'bug'},
+            ],
+        },
+    ];
+
+    const values = call.values;
+
+    // If values.issue_type is null, then we know that the user has just "cleared" this field. For our use case, we will just return the form as if the value was never selected.
+    let otherFields: AppField[] = [];
+
+    if (values.issue_type) {
+        if (values.issue_type.value === 'story') {
+            otherFields = getStoryFields();
+        } else if (values.issue_type.value === 'task') {
+            otherFields = getTaskFields();
+        } else if (values.issue_type.value === 'bug') {
+            otherFields = getBugFields();
+        }
+    }
+
+    // Combine the form's base fields with the new fields, to create the full form.
+    const allFields = baseFields.concat(otherFields);
+
+    const form: AppForm = {
+        title: 'Create a Ticket',
+        fields: allFields,
+    };
+
+    const call: AppCall = {
+        path: '/forms/create-ticket',
+        expand: {
+            acting_user: 'summary',
+            team: 'summary',
+        },
+    };
+
+    // Respond with modal form
+    const callResponse: AppCallResponse = {
+        type: 'form',
+        call,
+        form,
+    };
+
+    res.json(callResponse);
+});
+```
+
+Now we can respond to user action by rearranging the form's fields when the user selects an Issue Type. Note that this handler will be also called if the user clears the Issue Type field, or selects a new value after already choosing one. We can lock in the chosen value by providing a `readonly: true` property on the field after the user has selected it the first time.
```

Comment:
Making explanations on the comments and forcing the reader to read those, make the `Note that this handler will be also called if the user clears the Issue Type field` seems redundant. Having all the explanations in the same place may help the reading flow.